src/main/java/org/jenkinsci/plugins/github/pullrequest/GitHubPRTrigger.java

```diff
@@ -154,22 +149,21 @@ public String getRepoFullName() {
     }
 
     public String getRepoFullName(AbstractProject<?, ?> job) {
-        if (isNotBlank(repoFullName)) {
-            return repoFullName;
-        }
-
-        checkNotNull(job, "job object is null, race condition?");
-        GithubProjectProperty ghpp = job.getProperty(GithubProjectProperty.class);
+        if (isBlank(repoFullName)) {
+            
+            checkNotNull(job, "job object is null, race condition?");
+            GithubProjectProperty ghpp = job.getProperty(GithubProjectProperty.class);
 
-        checkNotNull(ghpp, "GitHub project property is not defined. Can't setup GitHub PR trigger for job %s", job.getName());
-        checkNotNull(ghpp.getProjectUrl(), "A GitHub project url is required");
+            checkNotNull(ghpp, "GitHub project property is not defined. Can't setup GitHub PR trigger for job %s", job.getName());
+            checkNotNull(ghpp.getProjectUrl(), "A GitHub project url is required");
 
-        String baseUrl = ghpp.getProjectUrl().baseUrl();
-        Matcher m = GH_FULL_REPO_NAME.matcher(baseUrl);
+            GitHubRepositoryName repo = GitHubRepositoryName.create(ghpp.getProjectUrl().baseUrl());
 
-        checkArgument(m.matches(), "Invalid GitHub project url: %s", baseUrl);
-
-        repoFullName = m.group(2);
+            checkNotNull(repo, "Invalid GitHub project url: %s", ghpp.getProjectUrl().baseUrl());
+            
+            repoFullName = String.format("%s/%s", repo.userName, repo.repositoryName);
```

Comment:
think it can be replaced with object reference instead of string in future. (If we wanna get multi-github support, full repo name should contain host and we get full object representation)